usmqe/web/tendrl/mainpage/clusters/cluster_list/pages.py

```diff
@@ -32,34 +63,81 @@ def start_create_cluster(self):
-            cluster_id (str): cluster id for choosing a cluster
-                              which should be imported
+            cluster_ident: cluster identificator for choosing a cluster
+                           which should be imported
             name (str): name of the cluster
                         TODO: Not used for now
                               https://github.com/Tendrl/api/issues/70
             hosts (list): list of dictionaries
                           {'hostname': <hostname>, 'release': <release>, ...
                           for check only
-                          TODO: not used for now
+
+        Returns:
+            tuple (cluster name or id, hosts list)
         """
 # TODO: Choose which cluster should be imported
 # TODO: Change cluster name
+        cluster_name = name
 #       https://github.com/Tendrl/api/issues/70
         self.start_import_cluster()
         import time
         import_page = ImportCluster(self.driver)
-# TODO: Check hosts list
+        cluster_id = import_page.cluster_id
+        if hosts is None:
+            release = import_page.storage_service
+            if 'Gluster' in release:
+                # get gluster hosts
+                host = next(iter(import_page.hosts)).name
+                storage = gluster.GlusterCommon()
+                hosts = [
+                    {'hostname': hostname, 'release': release, 'role': 'Peer'}
+                    for hostname in storage.get_hosts_from_trusted_pool(host)]
+            else:
+                # get ceph hosts
+                # TODO get the cluster name from somewhere
+                #       - configuration, cluster_id param or ...
+                cluster_name = cluster_name or 'test_name'
+                # NOTE there are no full hostnames available in ceph
+                monitors = []
+                for host in import_page.hosts:
+                    if host.role.lower() == 'monitor':
+                        monitors.append(host.name)
+                pytest.check(
+                    monitors != [],
+                    'There has to be a host with Monitor role in ceph cluster')
+                storage = ceph_cluster.CephCluster(cluster_name, monitors)
+                ceph_mons = storage.mon.stat()['mons'].keys()
+                mon_hosts = [
+                    {'hostname': hostname,
+                     'release': release,
+                     'role': 'Monitor'}
+                    for hostname in ceph_mons]
+                ceph_osds = []
+                ceph_all_osds = storage.osd.tree()['nodes']
+                for ceph_osd in ceph_all_osds:
+                    if ceph_osd['type'] == 'host':
+                        ceph_osds.append(ceph_osd['name'])
+                osds_hosts = [
+                    {'hostname': hostname,
+                     'release': release,
+                     'role': 'OSD Host'}
+                    for hostname in ceph_osds]
+                hosts = mon_hosts + osds_hosts
```

Comment:
Is it supposed to work with machine with both monitor and osd role? If not, it would be good to write this down somewhere or check for this situation in the code and report an error with explanation.